static/app/views/alerts/rules/issue/details/ruleDetails.spec.jsx

```diff
@@ -156,4 +158,42 @@ describe('AlertRuleDetails', () => {
       await screen.findByText('The alert rule you were looking for was not found.')
     ).toBeInTheDocument();
   });
+
+  it('renders incompatible rule filter', async () => {
+    const incompatibleRule = TestStubs.ProjectAlertRule({
+      lastTriggered: moment().subtract(2, 'day').format(),
+      conditions: [
+        {id: 'sentry.rules.conditions.first_seen_event.FirstSeenEventCondition'},
+        {id: 'sentry.rules.conditions.regression_event.RegressionEventCondition'},
+      ],
+    });
+    MockApiClient.mockResponses.splice(
+      MockApiClient.mockResponses.findIndex(
+        response =>
+          response.url ===
+          `/projects/${organization.slug}/${project.slug}/rules/${rule.id}/`
+      ),
+      1
+    );
+    MockApiClient.addMockResponse({
+      url: `/projects/${organization.slug}/${project.slug}/rules/${rule.id}/`,
+      body: incompatibleRule,
+      match: [MockApiClient.matchQuery({expand: 'lastTriggered'})],
+    });
+    createWrapper();
+    expect(
+      await screen.findByText(
+        'The conditions in this alert rule conflict and might not be working properly.'
+      )
+    ).toBeInTheDocument();
+  });
+
+  it('incompatible rule banner hidden for good rule', async () => {
```

Comment:
in RTL this doesn't really test anything since you can't wait for something to never appear

You could:
- wait for something else on the page that means that everything is loaded and then use queryByText
- consider unit testing the findIncompatibleRules directly

Reply:
I hope what i've done is what you mean by the first option.